src/views/studio/MyVideosView/MyVideosView.tsx

```diff
@@ -0,0 +1,189 @@
+// not yet doable
+// TODO: on edit video callbacks
+// TODO: on delete video callbacks
+// TODO: dynamic channels (not hardcoded)
+export const MyVideosView = () => {
+  const [videosPerRow, setVideosPerRow] = useState(INITIAL_VIDEOS_PER_ROW)
+  const [hasAnyVideos, sethasAnyVideos] = useState<boolean>()
+  // Drafts calls can run into race conditions
+  const { drafts, removeDraft, removeAllDrafts, addDraft } = useDrafts('video', testChannelId)
+  const [currentTab, setCurrentTab] = useState(0)
+  const { currentPage, setCurrentPage } = usePagination(currentTab)
+  const videosPerPage = 4 * videosPerRow
+  const currentTabName = TABS[currentTab]
+  const isPublic_eq = getPublicness(currentTabName)
+  const { loading, videos, error, totalCount, fetchMore } = useVideosOffsetLimitPagination(
+    {
+      limit: videosPerPage,
+      offset: videosPerPage * (currentPage - 1),
+      where: {
+        channelId_eq: testChannelId,
+        isPublic_eq,
+      },
+    },
+    {
+      // notifyOnNetworkStatusChange: true,
+      fetchPolicy: 'cache-first',
+    }
+  )
+
+  useEffect(() => {
+    if (typeof totalCount === 'number' && totalCount > 0) {
+      sethasAnyVideos(true)
+    } else if (typeof totalCount === 'number' && hasAnyVideos === undefined) {
+      sethasAnyVideos(false)
+    }
+  }, [hasAnyVideos, totalCount])
+
+  // hook to tests draft should be deleted before final
+  useEffect(() => {
+    const createDrafts = async () => {
+      await removeAllDrafts()
+      for (let i = 0; i < 32; i++) {
+        await addDraft({
+          channelId: testChannelId,
+          title: i.toString(),
+          description: 'string',
+          isPublic: true,
+          hasMarketing: false,
+          isExplicit: false,
+        })
+      }
+    }
+    createDrafts()
+    return () => {
+      removeAllDrafts()
+    }
+  }, [])
+
+  const isDraftTab = currentTabName === 'Drafts'
+  const isLoading = loading || (videos?.length === 0 && (totalCount ?? 0) > 0)
+  const placeholderItems = Array.from({ length: isLoading ? videosPerPage : 0 }, () => ({
+    id: undefined,
+    progress: undefined,
+  }))
+  const videosWPlaceholders = [...(videos || []), ...placeholderItems]
+  const handleOnResizeGrid = (sizes: number[]) => setVideosPerRow(sizes.length)
+
+  const FetchMoreVideos = (page: number) =>
+    fetchMore({
+      variables: {
+        // substract 1 coz offset index starts at 0
+        offset: videosPerPage * (page - 1),
+        limit: videosPerPage,
+      },
+    })
+  // console.log({ videos, totalCount, hasAnyVideos, isLoading, isPublic_eq })
+  return (
+    <ViewContainer>
+      <StyledText variant="h2">My Videos</StyledText>
+
+      {hasAnyVideos === false ? (
+        <AddVideoView />
+      ) : (
+        <>
+          <TabsContainer>
+            <Tabs initialIndex={0} tabs={[...TABS]} onSelectTab={setCurrentTab} />
+          </TabsContainer>
+          {isDraftTab && (
+            // Should this really be dismissable?
```

Comment:
What do you mean?

Reply:
I was thinking that it could be positives to the warning being always there as a reminder of the draft being only on local storage, but just a thought

Reply:
I think this is what the designers envisioned, let's keep it like this for now